WordPress/src/main/java/org/wordpress/android/ui/whatsnew/FeatureAnnouncementViewModel.kt

```diff
@@ -1,46 +1,73 @@
 package org.wordpress.android.ui.whatsnew
 
 import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
 import androidx.lifecycle.MutableLiveData
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
-import org.wordpress.android.WordPress
 import org.wordpress.android.modules.UI_THREAD
 import org.wordpress.android.viewmodel.ScopedViewModel
 import org.wordpress.android.viewmodel.SingleLiveEvent
 import javax.inject.Inject
 import javax.inject.Named
 
 class FeatureAnnouncementViewModel @Inject constructor(
+    private val featureAnnouncementProvider: FeatureAnnouncementProvider,
     @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
 ) : ScopedViewModel(mainDispatcher) {
-    private val _uiModel = MutableLiveData<FeatureAnnouncementUiModel>()
+    private val _currentFeatureAnnouncement = MutableLiveData<FeatureAnnouncement>()
+
+    private val _uiModel = MediatorLiveData<FeatureAnnouncementUiModel>()
     val uiModel: LiveData<FeatureAnnouncementUiModel> = _uiModel
 
     private val _onDialogClosed = SingleLiveEvent<Unit>()
     val onDialogClosed: LiveData<Unit> = _onDialogClosed
 
+    private val _onAnnouncementDetailsRequested = SingleLiveEvent<String>()
+    val onAnnouncementDetailsRequested: LiveData<String> = _onAnnouncementDetailsRequested
+
+    private val _featureItems = MediatorLiveData<List<FeatureAnnouncementItem>>()
+    val featureItems: LiveData<List<FeatureAnnouncementItem>> = _featureItems
+
     private var isStarted = false
 
+    init {
+        _uiModel.addSource(_currentFeatureAnnouncement) { featureAnnouncement ->
+            _uiModel.value = _uiModel.value?.copy(appVersion = featureAnnouncement.version, isProgressVisible = false)
+        }
+
+        _featureItems.addSource(_currentFeatureAnnouncement) { featureAnnouncement ->
+            _featureItems.value = featureAnnouncement.features
+        }
+    }
+
     fun start() {
         if (isStarted) return
         isStarted = true
 
-        _uiModel.value = FeatureAnnouncementUiModel(WordPress.versionName, isProgressVisible = true)
+        _uiModel.value = FeatureAnnouncementUiModel(isProgressVisible = true)
+
+        loadFeatures()
+    }
 
+    private fun loadFeatures() {
         launch {
             delay(3000)
-            _uiModel.value = _uiModel.value?.copy(isProgressVisible = false)
+            _currentFeatureAnnouncement.value = featureAnnouncementProvider.getLatestFeatureAnnouncement()
         }
     }
 
     fun onCloseDialogButtonPressed() {
         _onDialogClosed.call()
     }
 
+    fun onFindMoreButtonPressedPressed() {
```

Comment:
is this double `Pressed` intentional or a copy/paste thing?

Reply:
This is Copy/PastePaste thing :) Fixed!